### ISSUE TITLE
.WithDataVolume() on a Postgres resource results in authentication issues

### ISSUE THREAD

**Issue by thimok** [dotnet/aspire#4461]
In my Aspire AppHost, I defined a Postgres resource and one database to it. I also defined a service that has a reference to that database:

![image](https://github.com/dotnet/aspire/assets/20819818/b6b3326f-75f9-4d97-95e4-70a409dd554b)

Initially, I didn't have line 6 (WithDataVolume()) and the setup worked. However, this meant that all data was destroyed as soon as I stopped the application. This is undesireable, and I found the following documentation which suggested to use .WithDataVolume()::
https://learn.microsoft.com/en-us/dotnet/aspire/fundamentals/persist-data-volumes

![image](https://github.com/dotnet/aspire/assets/20819818/0ab38124-ccf3-418c-9235-d616d2322f36)

However, this seems to cause issues when the app starts, and it crashes immediately stating a password authentication failed:
```
Unhandled exception. Npgsql.PostgresException (0x80004005): 28P01: password authentication failed for user "postgres"
   at Npgsql.Internal.NpgsqlConnector.ReadMessageLong(Boolean async, DataRowLoadingMode dataRowLoadingMode, Boolean readingNotifications, Boolean isReadingPrependedMessage)
   at System.Runtime.CompilerServices.PoolingAsyncValueTaskMethodBuilder`1.StateMachineBox`1.System.Threading.Tasks.Sources.IValueTaskSource<TResult>.GetResult(Int16 token)
   at Npgsql.Internal.NpgsqlConnector.AuthenticateSASL(List`1 mechanisms, String username, Boolean async, CancellationToken cancellationToken)
   at Npgsql.Internal.NpgsqlConnector.Authenticate(String username, NpgsqlTimeout timeout, Boolean async, CancellationToken cancellationToken)
   at Npgsql.Internal.NpgsqlConnector.<Open>g__OpenCore|213_1(NpgsqlConnector conn, SslMode sslMode, NpgsqlTimeout timeout, Boolean async, CancellationToken cancellationToken, Boolean isFirstAttempt)
   at Npgsql.Internal.NpgsqlConnector.Open(NpgsqlTimeout timeout, Boolean async, CancellationToken cancellationToken)
   at Npgsql.UnpooledDataSource.Get(NpgsqlConnection conn, NpgsqlTimeout timeout, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlConnection.<Open>g__OpenAsync|42_0(Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlConnection.Open()
   at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenDbConnection(Boolean errorsExpected)
   at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenInternal(Boolean errorsExpected)
   at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.Open(Boolean errorsExpected)
   at Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal.NpgsqlDatabaseCreator.Exists(Boolean async, CancellationToken cancellationToken)
   at Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal.NpgsqlDatabaseCreator.Exists(Boolean async, CancellationToken cancellationToken)
   at Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal.NpgsqlDatabaseCreator.Exists()
   at Microsoft.EntityFrameworkCore.Migrations.HistoryRepository.Exists()
   at Microsoft.EntityFrameworkCore.Migrations.HistoryRepository.GetAppliedMigrations()
   at Npgsql.EntityFrameworkCore.PostgreSQL.Migrations.Internal.NpgsqlMigrator.Migrate(String targetMigration)
   at Microsoft.EntityFrameworkCore.RelationalDatabaseFacadeExtensions.Migrate(DatabaseFacade databaseFacade)
   at Program.<Main>$(String[] args) in D:\dev\[ProjectNameRedacted]\Code\Identity\[ProjectNameRedacted].Identity.Api\Program.cs:line 40
  Exception data:
    Severity: FATAL
    SqlState: 28P01
    MessageText: password authentication failed for user "postgres"
    File: auth.c
    Line: 323
    Routine: auth_failed
```

If I remove the .WithDataVolume() call, and restart the application host, it works again (but without persisting of course).

Packages used:
- Host: Aspire.Hosting.PostgreSQL, version 8.0.1.
- Service: Aspire.Npgsql.EntityFrameworkCore.PostgreSQL, version 8.0.1.

**Comment by DamianEdwards** [comment]
Yep, this is #1151 and we're aiming to fix in the upcoming 8.1 release.

**Comment by DamianEdwards** [comment]
Closing this now that #1151 is merged.
